Tasks/ExtractFiles/extractFiles.ts

```diff
@@ -0,0 +1,307 @@
+/// <reference path="../../definitions/vsts-task-lib.d.ts" />
+
+import fs = require('fs');
+import os = require('os');
+import path = require('path');
+import tl = require('vsts-task-lib/task');
+
+// archiveFilePatterns is a multiline input containing glob patterns
+var archiveFilePatterns: string[] = tl.getDelimitedInput('archiveFilePatterns', '\n', true);
+var destinationFolder :string  = path.normalize(tl.getPathInput('destinationFolder', true, false).trim());
+var cleanDestinationFolder: boolean = tl.getBoolInput('cleanDestinationFolder', false);
+
+var repoRoot : string = path.resolve(tl.getVariable('build.sourcesDirectory') || '');
+tl.debug('repoRoot: ' + repoRoot);
+
+var win = os.type().match(/^Win/);
+tl.debug('win: ' + win);
+
+// extractors
+var xpTarLocation : string;
+var xpUnzipLocation : string;
+// 7zip
+var xpSevenZipLocation: string;
+var winSevenZipLocation: string = '7zip/7z.exe';
+
+function getSevenZipLocation() : string {
+    if(win){
+        return winSevenZipLocation;
+    } else {
+        if (typeof xpTarLocation == "undefined") {
+            xpSevenZipLocation = tl.which('7z', true);
+        }
+        return xpSevenZipLocation; 
+    }
+}
+
+function findFiles(): string[] {
```

Comment:
findFiles() does use find() and match(); the rest of this is just to help the user with meaningful error messages when they get things wrong.